osu.Game/Rulesets/Scoring/HitResult.cs

```diff
@@ -225,12 +248,18 @@ public static bool IsHit(this HitResult result)
         /// <summary>
         /// Whether a <see cref="HitResult"/> is scorable.
         /// </summary>
-        public static bool IsScorable(this HitResult result) => result >= HitResult.Miss && result < HitResult.IgnoreMiss;
+        public static bool IsScorable(this HitResult result)
+        {
+            if (result == HitResult.LegacyComboIncrease)
+                return true;
```

Comment:
I'm slightly confused by this line, `LegacyComboIncrease` is marked as "scorable" here but it shouldn't affect the score? I can see multiple usages where non-scorable hit results are ignored on increasing combo, but I would imagine them to be adjusted instead? Asking here only from curiosity, no action needs to be taken if it's complicated.

Reply:
It's scorable primarily because the `ScoreProcessor` extraction functions pre-check `IsScorable()` for better or worse. I think it makes sense to be like this anyway since it does affect score - the combo portion of score to be exact.

> I can see multiple usages where non-scorable hit results are ignored on increasing combo

Are you referring to this? https://github.com/ppy/osu/blob/85705d97a532ffe732113ea224b8f77aa5db3c71/osu.Game/Rulesets/Scoring/ScoreProcessor.cs#L185-L190
That code is pretty weird. It's actually impossible for that to actually be the case because the only HitResults that can increase combo are guaranteed to have `IsScorable() == true`.

Reply:
Sounds like an inline comment alluding to this reasoning would be a nice addition.

Reply:
Just to clarify before I do anything, because I'm not 100% sure I understood the comment, is what I mentioned correct?

Reply:
Will need confirmation from @frenzibyte , but I'd expect something like this above the attached line:

`LegacyComboIncrease is not actually scorable, but needs to be defined as such to be correctly included in statistics output`.

Reply:
I'm pretty much referring to the fact that `LegacyComboIncrease` returns true on `IsScorable` while it doesn't directly increase the score. But I understand that it's done that way to not be ignored when increasing combo in `ScoreProcessor` and also to be included in the statistics output as @peppy mentioned as well. The suggested comment above will suffice.